router/router.go

```diff
@@ -1306,19 +1288,22 @@ func (rt *HandleT) findWorker(job *jobsdb.JobT, throttledAtTime time.Time) (toSe
 		// if yes returning worker only for 1 job
 		worker.abortedUserMutex.Lock()
 		defer worker.abortedUserMutex.Unlock()
-		if count, ok := worker.abortedUserIDMap[userID]; ok {
-			if count >= rt.allowAbortedUserJobsCountForProcessing {
+		if runningJobIDs, ok := worker.abortedUserIDMap[userID]; ok {
+			count := len(runningJobIDs)
+			_, hasJobID := runningJobIDs[job.JobID]
+
+			if !hasJobID && count >= rt.allowAbortedUserJobsCountForProcessing {
```

Comment:
@atzoum  Did you find any use for this variable? `allowAbortedUserJobsCountForProcessing`

Reply:
Well this is the cause of evil 😆 it is the concurrency limit imposed to a specific userID after a job has been aborted 

Can you please elaborate on your point?

Reply:
I have always felt this was a variable which is not necessary :( why do we need to impose a concurrency limit at a user level if his job is aborted?

Reply:
I didn't ever came to question this behaviour, so I suppose we have some good reason for doing it.
For sure this introduces a delay in processing for aborting users, but I suppose we are doing it for giving more room to successful jobs so that they can occupy most of the workers' time.

Reply:
Explanation for why `allowAbortedUserJobsCountForProcessing` was introduced.

This is done to prevent job status table blow up with waiting statuses. Assume generator is pickup jobs of users whose jobs are getting aborted after multiple retries. Every time a job is aborted, a max of 10K jobs will be pushed to the worker. Of these 10K jobs, first one will be failed and rest 9999 are going to be marked as waiting. 

To prevent this, if we find that a users jobs are getting aborted, we control how many jobs of that user are sent downstream to the worker for processing. By default, the value of the variable  `allowAbortedUserJobsCountForProcessing` is 1. Which means, only one job of the aborted user will be sent down to the worker for processing.

Reply:
Quick question: While the check `!hasJobID` has no harm in having. Is it really required?

Reply:
in a perfect world no this check wouldn't be required. This only guards us from forgetting to remove the jobID from the map, i.e. bug in the mechanism

Reply:
Another question: Isn't just the count of jobs in flight enough as opposed to keeping all the jobids? Not a big deal, just thinking.

Reply:
no, if you notice, there is a small detail. Once a job is aborted, we might have other jobs for the same userID already accepted in the pipeline. These jobs are not counted for, so they shouldn't decrement the counter. Practically the concurrency limiter applies for new jobs only

Reply:
Got it